packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -529,19 +543,26 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
 
     override render(): JSX.Element | undefined {
         const { hover } = this.state;
-        const { model, treeNode, colors, parentPath, sourceUri, decoration, labelProvider, commands, menus, contextKeys, caption } = this.props;
+        const { model, treeNode, colors, parentPath, sourceUri, decoration, labelProvider, commands, menus, contextKeys, caption, isLight } = this.props;
         const resourceUri = new URI(sourceUri);
 
+        const [light, dark] = [treeNode.decorations?.icon, treeNode.decorations?.iconDark];
+        const iconTheme = isLight ? light : dark;
+        const cName = typeof iconTheme === 'string' ? iconTheme : '';
+        const isURI = iconTheme && !cName;
```

Comment:
These checks are a little odd to read, we express a URI as something with an `iconTheme` and no `className`?

Reply:
Yeah, a name change was necessary.